packages/ui-components/src/components/QrCode/QrCodeVisual.tsx

```diff
@@ -92,58 +100,52 @@ export const QrCodeVisual = ({
   if (!isEnlarged) {
     if (options.length > 1) {
       return (
-        <Container className={className} maxWidth="sm">
-          <p style={{ color: 'white', textAlign: 'center' }}>QR Code</p>
+        <StyledContainer className={className} maxWidth="sm">
+          <StyledParagraph>QR Codes</StyledParagraph>
           {options
-            .filter((option, index) => index < 6)
+            .filter((option, index) => index < 4)
             .map(({ name, value }) => (
-              <Grid container style={{ paddingBottom: 5 }} alignContent="center">
-                <Grid item alignItems="flex-end" xs={6}>
-                  <QrCodeImage
-                    qrCodeContents={value}
-                    humanReadableId={name}
-                    width={50}
-                    margin="auto"
-                  />
-                </Grid>
-                <Grid item alignItems="flex-start" xs={6} style={{ justifyItems: 'center' }}>
-                  <Typography variant="body2" style={{ color: 'white' }}>
-                    {name}
-                  </Typography>
-                </Grid>
+              <Grid container direction="column">
+                <QrCodeImage
+                  qrCodeContents={value}
+                  humanReadableId={name}
+                  width={200}
+                  margin="auto"
+                  padding="0.5rem 0"
+                />
               </Grid>
             ))}
-        </Container>
+        </StyledContainer>
       );
     }
     const option = options[0];
     return (
-      <Container>
-        <p style={{ color: 'white', textAlign: 'center' }}>QR Code</p>
+      <StyledContainer>
```

Comment:
I think in both cases here there will be a title and a container. It might be a better idea then to create a `const` called something like `displayedQrCodes` and when there is more than 1, slice it to get the top 4, otherwise it would be the array with 1 item in there. Then loop through, and use a container around it. You can use `as` prop on the container to say if there is more than on it would be `Grid` otherwise `div`, e.g. `as={displayedQrCodes.length=== 1 ? 'div' : 'Grid'}`